core/src/main/java/de/jplag/highlightextraction/ContainedMatchesStrategy.java

```diff
@@ -30,4 +41,21 @@ public void processMatchTokenTypes(List<TokenType> matchTokenTypes, Consumer<Lis
         }
     }
 
+    /**
+     * Calculates the weight of a match considering subsequences of the match.
+     * @param match Considered match
+     * @param frequencyMap Frequency map build with processMatchTokenTypes method
+     * @param matchToken tokenType sequence of the match
+     * @return a weight for the match
+     */
+    @Override
+    public double calculateMatchFrequency(Match match, Map<List<TokenType>, Integer> frequencyMap, List<TokenType> matchToken) {
+        List<List<TokenType>> keys = getSubSequences(matchToken, minSubSequenceLength);
+        List<Integer> frequencies = new ArrayList<>();
+        for (List<TokenType> key : keys) {
+            frequencies.add(frequencyMap.getOrDefault(key, 0));
+        }
+        return frequencies.stream().filter(freq -> freq > 0).mapToInt(Integer::intValue).average().orElse(0.0);
```

Comment:
First you add "0" if not present. Then you filter them out. This can be refactored.